eth_account/signers/ledger.py

```diff
@@ -0,0 +1,196 @@
+
+        m / purpose' / coin_type' / account' / change / address_index
+
+        Apostrophe in the path indicates that bip32 hardened derivation is used.
+
+        Ledger expect the following bytes input:
+        1. Number of BIP 32 derivations to perform (max 10)
+        2. First derivation index (big endian)
+        3. Second derivation index (big endian)
+        ...
+        Nth. Last derivation index (big endian)
+
+        References:
+        - https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki
+        - https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki
+        '''
+        assert(path.startswith('m/'))
+
+        elements = path.split('/')[1:]
+        depth = len(elements)
+
+        result = bytes([depth])  # Number of BIP 32 derivations to perform (max 10)
+
+        for derivation_index in elements:
+            # For each derivation index in the path check if it is hardened
+            hardened = "'" in derivation_index
+            index = int(derivation_index.strip("'"))
+
+            if hardened:
+                # See bip32 spec for hardened derivation spec
+                index = 0x80000000 | index
+
+            # Append index to result as a big-endian (>) unsigned int (I)
+            result += struct.pack('>I', index)
+
+        return result
+
+    def _send_to_device(self, apdu):
+        if self.device is None:
+            self.device = ledgerblue.comm.getDongle(debug=False)
+
+        reply = self.device.exchange(apdu, timeout=60)
+
+        return reply
+
+    @property
+    def address(self):
+        return self.get_address(self.account_id)
+
+    def get_address(self, account_id):
+        '''
+        Query the ledger device for an ethereum address in HD wallet.
+        Offset is the number in the HD wallet tree
+        '''
+        bip32_path = self._path_to_bytes(self.path_prefix + str(account_id))
+
+        # https://github.com/LedgerHQ/blue-app-eth/blob/master/doc/ethapp.asc#get-eth-public-address
+        apdu = bytes.fromhex('e0020000')
+        apdu += bytes([len(bip32_path)])
+        apdu += bip32_path
+
+        result = self._send_to_device(apdu)
+
+        # Parse result
+        offset = 1 + result[0]
+        address = result[offset + 1: offset + 1 + result[offset]]
+        address = address.decode()  # Use decode() to convert from bytearray
+
+        return eth_utils.to_normalized_address(address)
+
+    def get_account_id(self, address, search_limit=20, search_account_id=0):
+        '''
+        Convert an address to the HD wallet tree account_id
+        Start search at an account_id. This would allow to search deeper if required.
+        Default search_limit at 20 take about 5s to reach.
+        '''
+        address = eth_utils.to_normalized_address(address)
+
+        for account_id in itertools.count(start=search_account_id):
+            if account_id > search_limit:
+                raise Exception(f'Address {address} not found' +
```

Comment:
Prefer use of `ValueError` or even a custom exception type here rather than a bare `Exception`